docs/architecture/grpc-for-wcf-developers/why-grpc.md

```diff
@@ -1,11 +1,45 @@
+
+## Similarity to WCF
+
+Although its implementation and approach is different, the actual experience of developing and consuming services with gRPC should be very intuitive for WCF developers. The underlying goal of making it possible to code as though the client and server were on the same platform, without needing to worry about networking, is the same. Both platforms share the principle of declaring and then implementing an interface, even though the process for declaring that interface is different. And as you'll see in chapter 5, the different types of RPC calls supported by gRPC map very well to the different bindings available to WCF services.
+
+## Benefits of gRPC
+
+Additional reasons why gRPC stands above other solutions are:
+
+### Performance
+
+As already discussed, using HTTP/2 rather than HTTP/1.1 removes the requirement for human-readable messages and instead uses the smaller faster binary protocol. This is more efficient for computers to parse. HTTP/2 also supports multiplexing requests over a single connection enabling responses to be sent as soon as they're ready without the need to wait in a queue (an issue in HTTP/1.1 known as "head-of-line (HOL) blocking"). Fewer resources are needed when using gRPC, which makes it a good solution to use for mobile devices and over slower networks.
+
+### Interoperability
+
+There are gRPC tools and libraries for all major programming languages and platforms, including .NET, Java, Python, Go, C++, Node.js, Swift, Dart, Ruby, and PHP. Thanks to the Protocol Buffers binary wire format and the efficient code generation for each platform, developers can achieve better performance than WCF's NetTCP bindings while still enjoying full cross-platform support.
+
+### Usability and productivity
+
+gRPC is a comprehensive RPC solution. It has thoroughly addressed the full range of networking and connectivity concerns. gRPC works consistently across multiple languages. gRPC provides excellent tooling, with much of the boilerplate auto-generated, so more developer time is freed up to focus on business logic.
+
+### Streaming
+
+gRPC has full bidirectional streaming, which is very similar to WCF's Full Duplex services but without the dependency on complicated network bindings and transports. gRPC streaming can operate over regular internet connections, load balancers, and service meshes.
+
+### Deadline/timeouts and cancellation
+
+gRPC allows clients to specify a maximum time for an RPC to complete. If the specified deadline is exceeded the server can cancel the operation independently of the client. Deadlines and cancellations can be propagated through further gRPC calls to help enforce resource usage limits. Clients may also abort operations when a deadline is exceeded, or earlier if necessary (e.g. due to a user interaction).
```

Comment:
This can be achieved with WCF using distributed transactions. When the deadline has been reached, the client aborts the transaction which will tell each server involved (including SQL server which gRPC can't propagate it's deadline to).

Reply:
We're not saying WCF can't do these things, just that gRPC can do them.